kotsadm/web/src/components/apps/AppVersionHistory.jsx

```diff
@@ -80,10 +88,75 @@ class AppVersionHistory extends Component {
     }
   }
 
+  componentDidUpdate = async (lastProps) => {
+    if (lastProps.match.params.slug !== this.props.match.params.slug || lastProps.app.id !== this.props.app.id) {
+      this.fetchKotsDownstreamHistory();
+    }
+  }
+
   componentWillUnmount() {
     this.state.updateChecker.stop();
   }
 
+  fetchKotsDownstreamHistory = async() => {
+    const { match, app } = this.props;
+    const downstream = app.downstreams[0];
+    const clusterSlug = downstream.cluster.slug;
+    const appSlug = match.params.slug;
+
+    this.setState({
+      loadingVersionHistory: true,
+      errorTitle: "",
+      errorMsg: "",
+      displayErrorModal: false,
+    });
+
+    try {
+      const res = await fetch(`${window.env.API_ENDPOINT}/app/${appSlug}/cluster/${clusterSlug}`, {
+        headers: {
+          "Authorization": Utilities.getToken(),
+          "Content-Type": "application/json",
+        },
+        method: "GET",
+      });
+      if (!res.ok) {
+        if (res.status === 401) {
+          Utilities.logoutUser();
+          return;
+        }
+        this.setState({
+          loadingVersionHistory: false,
+          errorTitle: "Failed to get version history",
+          errorMsg: `Unexpected status code: ${res.status}`,
+          displayErrorModal: true,
+        });
+        return;
+      }
+      const response = await res.json();
+      const versionHistory = response.versionHistory;
+
+      if (isAwaitingResults(versionHistory)) {
+        setTimeout(() => this.fetchKotsDownstreamHistory(), 2000);
```

Comment:
any suggestions to avoid this?

trying to replace this code https://github.com/replicatedhq/kots/pull/1010/files#diff-0104f4f9b1ac0ae7899cbddf9481296bL920-L923

Reply:
yes, use the `Repeater` class, and stop it when `isAwaitingResults` becomes false

Reply:
although what you have also works